File: src/main/java/javaCode/javadsl/LoginPage.java
package javaCode.javadsl;

import java.util.function.Consumer;

public class LoginPage {
    protected LoginPage clickBtn() {
        System.out.println("click btn");
        return this;
    }

    protected LoginPage sortItems() {
        System.out.println("sort items");
        return this;
    }

    protected LoginPage getText() {
        System.out.println("getText");
        return this;
    }

    protected LoginPage isLogoVisible() {
        System.out.println("isVisible");
        return this;
    }

    protected static LoginPage doActions(Consumer<LoginPage> init) {
        LoginPage actions = new LoginPage();
        init.accept(actions);
        return actions;
    }
}

File: src/main/java/kotlinCode/dsl/LoginPage.kt
package kotlinCode.dsl

open class LoginPage {
    fun clickBtn() {
        println("click btn")
    }

    protected fun sortItems() {
        println("sort items")
    }

    protected fun getText() {
        println("getText")
    }

    protected fun isLogoVisible() {
        println("isVisible")
    }

    fun doActions(init: LoginPage.() -> Unit): LoginPage {
        val actions = LoginPage()
        actions.init()
        return actions
    }

}